### PR TITLE
Closes #262, closes #270

### LINKED ISSUE #262: Publish via CI
**Is your feature request related to a problem? Please describe.**

I've moved a few projects like [Marten.FSharp](https://github.com/TheAngryByrd/Marten.FSharp/blob/master/.github/workflows/dotnet-publish.yml) and [IcedTasks](https://github.com/TheAngryByrd/IcedTasks/blob/master/.github/workflows/publish.yml) to publish from github actions.  I'm starting to prefer it and might be good to set as default.

**Describe the solution you'd like**

1. Create a publish.yml similar to the ones above
2. Need to update fake in the following ways:
  a. Change the set of targets to have a "Release" and a "Publish" target. Release will handle the Changelog and Git tag, Publish will handle the packaging, publish to nuget, and github release.
```fsharp
  
    "UpdateChangelog"
    ==> "GitRelease"
    ==>! "Release"



    "DotnetRestore"
    ==> "CheckFormatCode"
    ==> "DotnetBuild"
    // ==> "FSharpAnalyzers"
    ==> "DotnetTest"
    // =?> ("GenerateCoverageReport", not disableCodeCoverage)
    ==> "DotnetPack"
    ==> "PublishToNuGet"
    ==> "GitHubRelease"
    ==>! "Publish"
```
  b. fix the changelog diff link, as`linkReferenceForLatestEntry` won't be generated once,, need to be mutable anymore and can just be remade on demand in `mkReleaseNotes`
  
  ```fsharp
      let mkReleaseNotes (latestEntry : Changelog.ChangelogEntry) =
        let linkReference = mkLinkReference latestEntry.SemVer changelog
        if String.isNullOrEmpty linkReference then latestEntry.ToString()
        else
            // Add link reference target to description before building release notes, since in main changelog file it's at the bottom of the file
            let description =
                match latestEntry.Description with
                | None -> linkReference
                | Some desc when desc.Contains(linkReference) -> desc
                | Some desc -> sprintf "%s\n\n%s" (desc.Trim()) linkReference
            { latestEntry with Description = Some description }.ToString()
```
c. Fix prerelease changes

```fsharp
let prereleaseChanges = prereleaseEntries |> List.collect (fun entry -> entry.Changes |> List.filter (not << Changelog.isEmptyChange)) |> List.distinct
```

d. Create an `allPublishChecks` and update GithubRelease and PublishToNuget to use it

```fsharp

let allPublishChecks () =
    isOnCI ()
    Changelog.isChangelogEmpty ()

```

e. Figure out where docs fits in this this.

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #270: Replace docsTool with fsdocs.
**Is your feature request related to a problem? Please describe.**

`docsTool` has been out of date with FSharp.Formatting for a long time now.  FSharp.Formatting itself has gotten quite a bit of love since I originally made that tool. I think we should replace with docsTool with fsdocs.  

**Describe the solution you'd like**

I did the work over in IcedTasks to keep the structure of Tutorial/HowTo/Explanations/Reference.  

- [docsSrc](https://github.com/TheAngryByrd/IcedTasks/tree/master/docsSrc)
- [build helpers](https://github.com/TheAngryByrd/IcedTasks/blob/master/build/FsDocs.fs)
- [calling build helpers](https://github.com/TheAngryByrd/IcedTasks/blob/master/build/build.fs#L365)
- [publishing](https://github.com/TheAngryByrd/IcedTasks/blob/master/.github/workflows/fsdocs-gh-pages.yml)
- [published website](https://www.jimmybyrd.me/IcedTasks/)

**Describe alternatives you've considered**

Not doing it.  There's always an option not to.

**Additional context**

### PR DESCRIPTION
**Pull request by nikoyak**
## Proposed Changes

Almost line by line copying of the 'build' project and GitHub Actions from TheAngryByrd/IcedTasks repository. Similar changes have been made to the Content/Library template.

- Closes #262
- Closes #270 

## Types of changes

What types of changes does your code introduce to MiniScaffold?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] Build and tests pass locally
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Further comments

Tests had to be corrected.

### CODE REVIEW

**Comment by TheAngryByrd** [comment]
Thank you so much for this! I'll start reviewing it later tonight.

**Comment by nikoyak** [comment]
Sorry I didn't test CI build. Fixed.

File: Content/Library/build/Changelog.fs
Comment (TheAngryByrd) [inline Content/Library/build/Changelog.fs:1]: Could you revert the changes to Changelog.fs to keep it in a separate file? Might involve dealing with some unpleasant merging though.

**Review by TheAngryByrd — CHANGES_REQUESTED** [review]
Thank you so much for this!

One thing I'd like to revert/change listed in the feedback.

File: Content/Library/build/Changelog.fs
Comment (nikoyak) [reply Content/Library/build/Changelog.fs:1]: Done.

**Review by TheAngryByrd — APPROVED** [review]

**Pull request merged**